### PR TITLE
silx view: Added Windows installer generation

### PR DESCRIPTION
**Pull request by t20100**
This PR adds the config file and code to generate a Windows installer for silx view (silx as command line 
is also embedded).
The way to handle 2 executable in the same bundle is also updated since pyinstaller fixed an issue on this. It still requires a small hack but it is much better now.

This is copied/adapted from [crispy](https://github.com/mretegan/crispy),many thanks to @mretegan!!!
This saved me a lot of time!

### CODE REVIEW

**Comment by vallsv** [comment]
Sounds good.
Have you packaged it together with `h5plugin`? This could be really useful.

File: package/windows/pyinstaller.spec [hunk 1]
```diff
@@ -9,47 +13,131 @@ datas.append((os.path.join(PROJECT_PATH, "README.rst"), "."))
 datas.append((os.path.join(PROJECT_PATH, "LICENSE"), "."))
 datas.append((os.path.join(PROJECT_PATH, "copyright"), "."))
 datas += collect_data_files("silx.resources")
+datas += collect_data_files("hdf5plugin")
 
 
-hiddenimports = collect_submodules('fabio')
+hiddenimports = []
+hiddenimports += collect_submodules('fabio')
+hiddenimports += collect_submodules('hdf5plugin')
```
Comment (t20100) [inline package/windows/pyinstaller.spec:21]: @vallsv `hdf5plugin` is added here

Comment (vallsv) [reply package/windows/pyinstaller.spec:21]: Nice. Thx.

**Comment by t20100** [comment]
CI fails after tests pass (see #3553)

File: package/windows/pyinstaller.spec (hunk 1 above)
Comment (mretegan) [reply package/windows/pyinstaller.spec:21]: Should `hdf5plugin` be added to the requirements-dev.txt?

**Comment by mretegan** [comment]
I've built the binary on a Windows 7 VM and installed it on another with Windows 10. Unfortunately, it doesn't work, and I get the following error: "Fatal error detected. Archive not found: C:\Program Files\silx\bootstrap."

File: package/windows/pyinstaller.spec (hunk 1 above)
Comment (t20100) [reply package/windows/pyinstaller.spec:21]: Yes, indeed!

**Comment by t20100** [comment]
Installer built with CI is available here: https://gitlab.esrf.fr/silx/bob/silx/-/jobs/275225/artifacts/raw/artifacts/silx-1.0.0-a0-x64.exe

**Comment by mretegan** [comment]
Using the binary generated by the CI everything works as expected.

**Comment by mretegan** [comment]
btw `hdf5plugin` is still missing from `requirements-dev.txt`.

**Pull request merged**